engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -112,7 +112,8 @@ Object doFunctionalDispatch(
       Object self,
       Object[] arguments,
       @CachedLibrary(limit = "10") TypesLibrary typesLibrary,
-      @Cached MethodResolverNode methodResolverNode) {
+      @Cached MethodResolverNode methodResolverNode,
+      @Cached("anyFunction(symbol)") Function anyFun) {
```

Comment:
I think that you should also cache `symbol` here, and check whether it is still same in the guards. In this form, the `anyFunc` is resolved during the *specialization instantiation*, but you can potentially get a different symbol every time.

My suggestion is: Add `@Cached("symbol") UnresolvedSymbol cachedSymbol` to the parameters, and `"cachedSymbol.equals(symbol)"` (or `"cachedSymbol == symbol"`) to the guards.